fix(minifier): inline multiple variable declarations at once

sapphi-red opened the pull request:
Codes like:
```js
var a1 = 'a1'
var a2 = 'a2'
var a3 = 'a3'
var arr = [
  a1,
  a2,
  a3,
]
console.log(arr)
```
was not processed at once for DCE because DCE does not join variable declarations.
This PR fixes that.

Boshen commented:
### Merge activity

* **Aug 31, 3:03 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
fix(minifier): inline multiple variable declarations at once (#13477)

Codes like:
```js
var a1 = 'a1'
var a2 = 'a2'
var a3 = 'a3'
var arr = [
  a1,
  a2,
  a3,
]
console.log(arr)
```
was not processes at once for DCE because DCE does not join variable declarations.
This PR fixes that.
